userapi/src/main/java/model/D42Model.java

```diff
@@ -0,0 +1,117 @@
+package model;
+
+
+import fk.prof.storage.AsyncStorage;
+import fk.prof.storage.S3AsyncStorage;
+import org.apache.commons.codec.binary.Base32;
+import org.slf4j.LoggerFactory;
+
+import java.time.ZonedDateTime;
+import java.util.HashSet;
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.stream.Collectors;
+
+/**
+ * Interacts with the AsyncStorage based on invocations from controller
+ * Created by rohit.patiyal on 19/01/17.
+ */
+public class D42Model implements IDataModel {
+    private static final String ACCESS_KEY = "66ZX9WC7ZRO6S5BSO8TG";
```

Comment:
Same here. We should rename D42Model. I dont see any code here that is directly dependent on storage implementation.